include/ipr/traversal

```diff
@@ -84,6 +85,26 @@ namespace ipr {
          n.accept(vis);
          return vis.result;
       }
+
+      // Similar function to above, but get the implementation 
+      // node instead of the interface node.
+      template<class T>
+      typename impl::traits<T>::rep*
+      view_impl(Node& n)
+      {
+         struct visitor : Constant_visitor<No_op> {
+            typename impl::traits<T>::rep* result;
+            visitor() : result() { }
+            void visit(const T& n) override {
+               result = static_cast<typename impl::traits<T>::rep*>(
```

Comment:
As far as I understand:
- Static cast should be safe, since we have a linearized inheritance chain.
- const_cast should be safe, as the non-const reference argument of `view_impl` guarantees that the node we have was not const to begin with.